javascript/ql/src/semmle/javascript/frameworks/Electron.qll

```diff
@@ -73,187 +73,109 @@ module Electron {
   /**
    * A reference to the `webContents` property of a browser object.
    */
-  class WebContents extends DataFlow::SourceNode {
+  class WebContents extends DataFlow::SourceNode, EventEmitter::EventEmitterRange::NodeJSEventEmitter {
     WebContents() { this.(DataFlow::PropRead).accesses(any(BrowserObject bo), "webContents") }
   }
 
   /**
    * Provides classes and predicates for modelling Electron inter-process communication (IPC).
+   * The Electron IPC are EventEmitters, but they also expose a number of methods on top of the standard EventEmitter.
    */
   private module IPC {
-    class Process extends string {
-      Process() { this = "main" or this = "renderer" }
+    DataFlow::SourceNode main() { result = DataFlow::moduleMember("electron", "ipcMain") }
 
-      DataFlow::SourceNode getAnImport() {
-        this = Process::main() and result = DataFlow::moduleMember("electron", "ipcMain")
-        or
-        this = Process::renderer() and result = DataFlow::moduleMember("electron", "ipcRenderer")
-      }
-    }
-
-    module Process {
-      Process main() { result = "main" }
-
-      Process renderer() { result = "renderer" }
-    }
-
-    /**
-     * An IPC callback.
-     */
-    class Callback extends DataFlow::FunctionNode {
-      DataFlow::Node channel;
-      Process process;
-
-      Callback() {
-        exists(DataFlow::MethodCallNode mc |
-          mc = process.getAnImport().getAMemberCall("on") and
-          this = mc.getCallback(1) and
-          channel = mc.getArgument(0)
-        )
-      }
-
-      /** Gets the process on which this callback is executed. */
-      Process getProcess() { result = process }
-
-      /** Gets the name of the channel the callback is listening on. */
-      string getChannelName() { result = channel.getStringValue() }
-
-      /** Gets the data flow node containing the message received by the callback. */
-      DataFlow::Node getMessage() { result = getParameter(1) }
-    }
+    DataFlow::SourceNode renderer() { result = DataFlow::moduleMember("electron", "ipcRenderer") }
 
     /**
-     * An IPC message.
+     * A model for the Main and Renderer process in an Electron app.
      */
-    abstract class Message extends DataFlow::Node {
-      /** Gets the process that sends this message. */
-      abstract Process getProcess();
-
-      /** Gets the name of the channel this message is sent on. */
-      abstract string getChannelName();
+    abstract class Process extends EventEmitter::EventEmitterRange::Range {
+      /**
+       * Type tracking on a process. The type tracking tracks through chainable methods.
```

Comment:
This comment doesn't follow our usual convention; it should start with "Gets ...".